search/meilisearch.py

```diff
@@ -0,0 +1,389 @@
+        filter_query_field = {
+            field_name: field_value,
+        }
+    return filter_query_field
+
+
+def _process_field_queries(field_dictionary):
+    """
+    Prepare Meilisearch query which must be in the Meilisearch record set.
+    """
+    return [
+        _get_filter_field(field, field_value)
+        for field, field_value in field_dictionary.items()
+    ]
+
+
+def _process_filters(filter_dictionary):
+    """
+    Build list for filtering.
+    """
+    for field, value in filter_dictionary.items():
+        if value:
+            yield _get_filter_field(field, value)
+
+
+def _process_exclude_dictionary(exclude_dictionary):
+    """
+    Build a list of term fields which will be excluded from result set.
+    """
+    for exclude_property, exclude_values in exclude_dictionary.items():
+        if not isinstance(exclude_values, list):
+            exclude_values = (exclude_values,)
+        yield from (
+            {exclude_property: exclude_value}
+            for exclude_value in exclude_values
+        )
+
+
+def _process_aggregation_terms(aggregation_terms):
+    """
+    Meilisearch does not support aggregations natively as Elasticsearch.
+    """
+    return aggregation_terms
+
+
+class MeiliSearchEngine(SearchEngine):
+    """
+    Meilisearch implementation of SearchEngine abstraction
+    """
+    backend_name = "meilisearch"
+
+    @staticmethod
+    def get_cache_item_name(index_name):
+        """
+        Name-formatter for cache_item_name
+        """
+        return f"meili_search_mappings_{index_name}"
+
+    @classmethod
+    def get_mappings(cls, index_name):
+        """
+        Fetch mapped-items structure from cache
+        """
+        return cache.get(cls.get_cache_item_name(index_name), {})
+
+    @classmethod
+    def set_mappings(cls, index_name, mappings):
+        """
+        Set new mapped-items structure into cache
+        """
+        cache.set(cls.get_cache_item_name(index_name), mappings)
+
+    @property
+    def mappings(self):
+        """
+        Get mapping of current index.
+
+        Mappings format in Meilisearch is different from Elasticsearch.
+        """
+        mapping = MeiliSearchEngine.get_mappings(self._prefixed_index_name)
+        if not mapping:
+            # Assuming Meilisearch mappings are pre-defined elsewhere
+            mapping = {}  # Update this if there's a way to fetch mappings
+            if mapping:
+                MeiliSearchEngine.set_mappings(self._prefixed_index_name, mapping)
+        return mapping
+
+    def _clear_mapping(self):
+        """
+        Remove the cached mappings.
+        """
+        MeiliSearchEngine.set_mappings(self._prefixed_index_name, {})
+
+    def __init__(self, index=None, options=None):
+        super().__init__(index)
+        MEILISEARCH_URL = getattr(settings, "MEILISEARCH_URL", 'http://127.0.0.1:7700')
+        MEILISEARCH_API_KEY = getattr(settings, "MEILISEARCH_API_KEY", "masterKey")
```

Comment:
I don't know if we should be putting the default values here - we might get inconsistent defaults if we just define the defaults separately in each place where we read the setting.